src/plots/cartesian/axes.js

```diff
@@ -2305,16 +2327,37 @@ axes.drawOne = function(gd, ax, opts) {
 
     // this happens when updating matched group with 'missing' axes
     if(!mainPlotinfo) return;
+    // Will this axis 'push' out other axes?
+    ax._shiftPusher = overlayingShiftedAx.includes(ax._id) || overlayingShiftedAx.includes(ax.overlaying) || ax.shift === true;
+    // An axis is also shifted by 1/2 of its own linewidth and inside tick length if applicable
+    if(ax._shiftPusher & ax.anchor === 'free') {
+        var selfPush = (ax.linewidth / 2 || 0);
+        if(ax.ticks === 'inside') {
+            selfPush += ax.ticklen;
+        }
+        axShifts = incrementShift(ax, selfPush, axShifts);
+    }
+
+    // Somewhat inelegant way of making sure that the shift value is only updated when the
+    // Axes.DrawOne() function is called from the right context. An issue when redrawing the
+    // axis as result of using the dragbox, for example.
+    if(opts.skipTitle === true) {
+        ax._shift = ax._shift === undefined ? setShiftVal(ax, axShifts) : ax._shift;
+    } else {
+        ax._shift = setShiftVal(ax, axShifts);
+    }
```

Comment:
It looks like we could combine two if statements above into one line something like: 
```js
    if(opts.skipTitle !== true || ax._shift === undefined) ax._shift = setShiftVal(ax, axShifts);
```